crates/runtime/src/logic.rs

```diff
@@ -227,9 +239,7 @@ impl VMHostFunctions<'_> {
 
         String::from_utf8(buf).map_err(|_| HostError::BadUTF8.into())
     }
-}
 
-impl VMHostFunctions<'_> {
```

Comment:
this is semantic separation, please restore it

the above are utility methods, not host functions

and below are host functions

Reply:
Apologies - it's a little unorthodox and so I thought it was a mistake 😅 Restored! 👍